targets/DeployDependencies.proj

```diff
@@ -25,20 +26,32 @@
   <PropertyGroup>
     <RuntimeProjectJson>$(MSBuildThisFileDirectory)\runtime.project.json</RuntimeProjectJson>
     <RuntimeProjectLockJson>$(MSBuildThisFileDirectory)\runtime.project.lock.json</RuntimeProjectLockJson>
-    <RestoreRuntimePackagesCommand>"$(NuGetToolPath)" restore $(RuntimeProjectJson)</RestoreRuntimePackagesCommand>
+    <RestoreRuntimePackagesCommand>$(NugetRestoreCommand) $(NuGetPackageInstallOptions) $(RuntimeProjectJson)</RestoreRuntimePackagesCommand>
     <RuntimeDirectory Condition="'$(RuntimeDirectory)' ==''">$(OutputPath)</RuntimeDirectory>
+    <!-- Put $(RuntimeStamp) in $(RuntimeDir) to make deleting the directory
+         do the right thing.
+    -->
+    <RuntimeStamp>$(RuntimeDirectory)\runtime.stamp</RuntimeStamp>
     <NuGetRuntimeIdentifier>$(RuntimeSystem)-$(RuntimeArchitecture)</NuGetRuntimeIdentifier>
   </PropertyGroup>
 
   <Target Name="RestoreRuntimePackages"
           Inputs="$(NuGetToolPath);$(RuntimeProjectJson)"
-          Outputs="$(RuntimeProjectLockJson)"
+          Outputs="$(RuntimeStamp)"
           >
+
+    <Message Importance="High" Text="Restoring NuGet packages..." />
     <Exec Command="$(RestoreRuntimePackagesCommand)" StandardOutputImportance="Low" />
 
-    <!-- It appears that NuGet doesn't re-write the project.lock.json file if the contents would be the same. This can mess up incremental builds
-        if you make a non-significant change to project.json.  To avoid that, update the timestamp of the lock file here. -->
-    <Touch Files="$(RuntimeProjectLockJson)" />
+    <!-- This is just a "stamp" file, serving no purpose besides indicating
+         that the target has been run since $(RuntimeProjectJson) was last
+         updated.  The reason we can't use $(RuntimeProjectLockJson) for this
+         is that, since $(RuntimeProjectJson) and $(RuntimeProjectLockJson)
+         are both under version control, their timestamps are left in an
+         indeterminate order by various operations, such as "git clone",
+         "git pull", and "git checkout".
```

Comment:
I think this is a solid, convincing reason to use a separate file as the semaphore.